source/common/common/key_value_store_base.cc

```diff
@@ -0,0 +1,102 @@
+#include "source/common/common/key_value_store_base.h"
+
+namespace Envoy {
+namespace {
+
+// Removes a length prefixed token from |contents| and returns the token,
+// or returns absl::nullopt on failure.
+absl::optional<absl::string_view> getToken(absl::string_view& contents) {
+  const auto it = contents.find("\n");
+  if (it == contents.npos) {
+    ENVOY_LOG_MISC(warn, "Bad file: no newline");
```

Comment:
Should these use a logger id instead of misc?

Reply:
yeah the loggers don't very play well with helper functions.  I can make the base function a member function instead and have the helpers return up errors though.